File: BOJ_1931.kt
fun main(){
    val N: Int = readLine()!!.toInt()
    val list = mutableListOf<Pair<Int, Int>>()

    for(i in 0 until N){
        list.add(Pair(readLine()!!.toInt(), readLine()!!.toInt()))
    }

    list.sortBy {
        it.second
    }
    println(list)

    var a: Int
    var result = 1

    a = list[0].second
    for(i in 0 until list.size-1){
        if(a<=list[i+1].first){
            a = list[i+1].second
            result++
        }else{
            continue
        }
    }
    println(result)
}

File: BOJ_1931.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const pair<int, int> a, const pair<int, int> b){
    if(a.second==b.second){
        return a.first<b.first;
    }else{
        return a.second<b.second;
    }
}

int main(){
    int N, f, s, cnt=1;
    cin >> N;
    vector<pair<int, int>> vec;

    for(int i=0; i<N; i++){
        cin >> f >> s;
        vec.push_back(make_pair(f, s));
    }

    sort(vec.begin(), vec.end(), cmp);
    
    int ret = vec[0].second;

    for(int i=0; i<N-1; i++){
        if(ret<=vec[i+1].first){
            ret = vec[i+1].second;
            cnt++;
        }else{
            continue;
        }
    }
    cout << cnt;
}